### PR TITLE
Add UWP implementation for SelectAllTextEffect

### PR DESCRIPTION
**Pull request by Alex-Witkowski**
### Description of Change ###

Adds implementation of SelectAllTextEffect for UWP platform. Effect is based on TextBox which is base class of FormsTextBox used for Entry and Edit. 

### Bugs Fixed ###

Sorry no Issue created (yet). 

### API Changes ###

n.a.

### Behavioral Changes ###

Effect now just works with UWP. 

### PR Checklist ###

- [ ] Has tests (if omitted, state reason in description) => Don't see this as practical but I'm here to learn.
- [x] Has samples (if omitted, state reason in description)
- [x] Rebased on top of main at time of PR => Rebased on current Develop
- [x] Changes adhere to coding standard
- [ ] Updated [documentation](https://github.com/MicrosoftDocs/xamarin-communitytoolkit) => ToDo

### CODE REVIEW

**Comment by Alex-Witkowski** [comment]
Regarding documentation - is it possible to create an overview with information what effect is supported on which platform? Does this make sens?

File: src/CommunityToolkit/Xamarin.CommunityToolkit/Effects/SelectAllText/SelectAllTextEffect.uwp.cs [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+		protected override void OnAttached()
+		{
+			ApplyEffect(true);
+		}
+
+		protected override void OnDetached()
+		{
+			ApplyEffect(true);
+		}
```
Comment (AndreiMisiukevich) [inline src/CommunityToolkit/Xamarin.CommunityToolkit/Effects/SelectAllText/SelectAllTextEffect.uwp.cs:25]:
```suggestion
		protected override void OnAttached() => ApplyEffect(true);

		protected override void OnDetached() => ApplyEffect(true);
```

File: src/CommunityToolkit/Xamarin.CommunityToolkit/Effects/SelectAllText/SelectAllTextEffect.uwp.cs [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+			if (apply)
+				EditText.GotFocus += OnGotFocus;
+			else
+				EditText.GotFocus -= OnGotFocus;
+		}
+
+		void OnGotFocus(object sender, RoutedEventArgs e)
+		{
+			EditText?.SelectAll();
+		}
```
Comment (AndreiMisiukevich) [inline src/CommunityToolkit/Xamarin.CommunityToolkit/Effects/SelectAllText/SelectAllTextEffect.uwp.cs:44]:
```suggestion
		void OnGotFocus(object sender, RoutedEventArgs e) => EditText?.SelectAll();
```

File: src/CommunityToolkit/Xamarin.CommunityToolkit/Effects/SelectAllText/SelectAllTextEffect.uwp.cs [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+			{
+				throw new NotSupportedException(
+					$"Control of type: {Control?.GetType()?.Name} is not supported by this effect.");
+			}
+
+			if (apply)
+				EditText.GotFocus += OnGotFocus;
+			else
+				EditText.GotFocus -= OnGotFocus;
+		}
```
Comment (AndreiMisiukevich) [inline src/CommunityToolkit/Xamarin.CommunityToolkit/Effects/SelectAllText/SelectAllTextEffect.uwp.cs:39]:
```suggestion
		{
			if (EditText == null)
				throw new NotSupportedException(
					$"Control of type: {Control?.GetType()?.Name} is not supported by this effect.");
					
			EditText.GotFocus -= OnGotFocus;
			
			if (apply)
				EditText.GotFocus += OnGotFocus;
		}
```

**Comment by pictos** [comment]
> Regarding documentation - is it possible to create an overview with information what effect is supported on which platform? Does this make sens?

@Alex-Witkowski we don't have this effect documented, so will be great if you can do it! You can use the SafeAreaEffect as a sample.

**Review by AndreiMisiukevich — APPROVED** [review]

**Comment by AndreiMisiukevich** [comment]
@Alex-Witkowski can you please rebase on the head of develop?

**Review by jfversluis — APPROVED** [review]

**Review by AndreiMisiukevich — APPROVED** [review]

**Pull request merged**